pulsar-zookeeper/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperServerAspect.java

```diff
@@ -31,15 +31,26 @@
  */
 @Aspect
 public class ZooKeeperServerAspect {
+    private static boolean metricsRegistered = false;
+
     @Pointcut("execution(org.apache.zookeeper.server.ZooKeeperServer.new(..))")
-    public void processRequest() {
+    public void zkServerConstructorPointCut() {
     }
 
-    @After("processRequest()")
-    public void timedProcessRequest(JoinPoint joinPoint) throws Throwable {
+    @After("zkServerConstructorPointCut()")
+    public void zkServerConstructor(JoinPoint joinPoint) throws Throwable {
         // ZooKeeperServer instance was created
         ZooKeeperServer zkServer = (ZooKeeperServer) joinPoint.getThis();
 
+        synchronized (ZooKeeperServerAspect.class) {
+            if (metricsRegistered) {
```

Comment:
it's not double checked, since the mutex it's upfront :) 

double checked would be trying to avoid the mutex in the best case, like: 

```java
if (metricsRegistered) {
   return;
} else {
    synchronized (ZooKeeperServerAspect.class) {
        if (metricsRegistered) {
           return;
        }
    }
}
```